tests/Console/Command/DescribeCommandTest.php

```diff
@@ -247,11 +406,62 @@ public function testCommandDescribesCustomFixer(): void
         self::assertSame(0, $commandTester->getStatusCode());
     }
 
-    private function execute(string $name, bool $decorated): CommandTester
+    /**
+     * @param CodeSampleInterface[] $samples
+     */
+    private static function getMockedFixerWithSamples(array $samples): FixerInterface
+    {
+        return new class($samples) extends AbstractFixer {
+            /**
+             * @var CodeSampleInterface[]
+             */
+            private $samples;
+
+            /**
+             * @param CodeSampleInterface[] $samples
+             */
+            public function __construct(
+                array $samples
+            ) {
+                parent::__construct();
+                $this->samples = $samples;
+            }
+
+            public function getName(): string
+            {
+                return 'Foo/samples';
+            }
+
+            public function getDefinition(): FixerDefinitionInterface
+            {
+                return new FixerDefinition(
+                    'Summary of the rule.',
+                    $this->samples,
+                    'Description of the rule.',
+                    null,
+                );
+            }
+
+            public function isCandidate(Tokens $tokens): bool
+            {
+                return true;
+            }
+
+            public function applyFix(\SplFileInfo $file, Tokens $tokens): void
+            {
+                $tokens[3] = new Token([
+                    $tokens[3]->getId(),
+                    "'AFTER'",
+                ]);
```

Comment:
Since `VersionSpecificCodeSample` samples are provided from the outside, the contract for applying fix is really loose here, basically it always need to be `<?php echo 'BEFORE'` and then something more. Wouldn't it make sense to make some lop here and detect all the `'BEFORE'` strings in `$tokens`? It's good enough to go as-is, though.

Reply:
if you want, you can even create a real, proper fixer class here. Yet, this test is not aiming to test Fixer class implementation, i would even go with bare text regex and if we wouldn't use same tokens instance reference.